homeassistant/components/switchbot_cloud/__init__.py

```diff
@@ -0,0 +1,63 @@
+"""The SwitchBot via API integration."""
+from asyncio import gather
+from dataclasses import dataclass
+from logging import getLogger
+
+from switchbot_api import Device, Remote, SwitchBotAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_API_TOKEN, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import SwitchBotCoordinator
+
+_LOGGER = getLogger(__name__)
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+
+@dataclass
+class Data:
+    """Data to use in platforms."""
+
+    api: SwitchBotAPI
+    switches: list[Device | Remote]
+
+
+async def async_setup_entry(hass: HomeAssistant, config: ConfigEntry) -> bool:
+    """Set up SwitchBot via API from a config entry."""
+    token = config.data[CONF_API_TOKEN]
+    secret = config.data[CONF_API_KEY]
+
+    api = SwitchBotAPI(token=token, secret=secret)
+    devices = await api.list_devices()
+    _LOGGER.debug("Devices: %s", devices)
+    devices_and_coordinators = [
+        (device, SwitchBotCoordinator(hass, api, device)) for device in devices
+    ]
+    hass.data.setdefault(DOMAIN, {})
+    data = Data(
+        api=api,
+        switches=[
```

Comment:
```suggestion
        devices=[
```

Maybe call this `devices` since I expect it will support more than just switches in the future

Reply:
@bdraco I was thinking about making a different list for each platform

Reply:
That sounds good. Maybe even a new dataclass inside this one?

Reply:
I am sorry, I do not understand the purpose of a new dataclass inside this one.

I was think about something like:
```python
class SwitchbotCloudData:
    """Data to use in platforms."""

    api: SwitchBotAPI
    climates: list[Remote]
    lights: list[Device | Remote]
    switches: list[Device | Remote]
    ...
```

Reply:
I was thinking something like this

```

class SwitchbotDevices:
    """Switchbot devices data."""

    climates: list[Remote]
    lights: list[Device | Remote]
    switches: list[Device | Remote]



class SwitchbotCloudData:
    """Data to use in platforms."""

    api: SwitchBotAPI
    devices: SwitchbotDevices
```

Reply:
I expect the list of device types will grow over time and it will keep the main data class from getting a bit unwieldy